logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/folders/ConversationFolderRepository.kt

```diff
@@ -119,4 +125,36 @@ internal class ConversationFolderDataSource internal constructor(
         }
         .map { }
 
+    override suspend fun addConversationToFolder(conversationId: QualifiedID, folderId: String): Either<CoreFailure, Unit> {
+        kaliumLogger.withFeatureId(CONVERSATIONS_FOLDERS)
+            .v("Adding conversation ${conversationId.toLogString()} to folder ${folderId.obfuscateId()}")
+        return wrapStorageRequest {
+            conversationFolderDAO.addConversationToFolder(conversationId.toDao(), folderId)
+        }
+            .flatMap {
+                syncConversationFoldersFromLocal()
+            }
+    }
+
+    override suspend fun removeConversationFromFolder(conversationId: QualifiedID, folderId: String): Either<CoreFailure, Unit> {
+        kaliumLogger.withFeatureId(CONVERSATIONS_FOLDERS)
+            .v("Removing conversation ${conversationId.toLogString()} from folder ${folderId.obfuscateId()}")
+        return wrapStorageRequest {
+            conversationFolderDAO.removeConversationFromFolder(conversationId.toDao(), folderId)
+        }.flatMap {
+            syncConversationFoldersFromLocal()
+        }
+    }
+
+    override suspend fun syncConversationFoldersFromLocal(): Either<CoreFailure, Unit> {
+        kaliumLogger.withFeatureId(CONVERSATIONS_FOLDERS).v("Syncing conversation folders from local")
+        return wrapStorageRequest { conversationFolderDAO.getFoldersWithConversations().map { it.toModel() } }
+            .flatMap {
+                wrapApiRequest {
+                    userPropertiesApi.updateLabels(
+                        LabelListResponseDTO(it.map { it.toLabel() })
+                    )
+                }
+            }
+    }
```

Comment:
This is also breaking the separation of concerns.

We can have this logic in a use case, as it's responsible for orchestrating business logic.